geometry.go

```diff
@@ -93,6 +93,14 @@ func (u Vec) Unit() Vec {
 	return u.Scaled(1 / u.Len())
 }
 
+// ScaledTo returns a vector of specified length facing the direction of u (has the same angle).
+func (u Vec) ScaledTo(len float64) Vec {
+	if u.X == 0 && u.Y == 0 {
+		return Vec{len, 0}
+	}
+	return u.Scaled(len / u.Len())
+}
+
```

Comment:
What's the performance impact of this? If it's negligible, I'd prefer `.Unit().Mul(len)` instead (less API).